docs/network-policy.md

```diff
@@ -0,0 +1,144 @@
+selected by the namespaceSelector will be selected. Specific Pods from
+specific Namespaces can be selected by using a combination of
+`podSelector` and `namespaceSelector` in the same peer. IPBlock is not allowed
+to be set in the `appliedTo` field.
+In the example, the policy applies to Pods, which either match the labels
+"role=db" in all the Namespaces, or are from Namespaces which match the
+labels "env=prod".
+
+**priority**: The `priority` field determines the relative priority of the policy
+among all ClusterNetworkPolicies in the given cluster. This field is mandatory.
+A lower priority value indicates higher precedence. Priority values can range
+from 1.0-10000.0.
+**Note**: Policies with same priorities will be evaluated
+indeterministically. Users should therefore take care to use priorities to
+ensure the behavior they expect.
+
+**ingress**: Each ClusterNetworkPolicy may consist of zero or more ordered
+set of ingress rules. Each rule, depending on the `action` field of the rule,
+allows or drops traffic which matches both the `from` and `ports` sections.
+The example policy contains a single rule, which allows matched traffic on a
+single port, from one of the two sources, first specified by the `podSelector`
+and second, specified by the combination of `podSelector` and
+`namespaceSelector`.
+**Note**: The order in which the ingress rules are set matter. i.e. rules will be
+evaluated in the order in which they are written.
+
+**egress**: Each ClusterNetworkPolicy may consist of zero or more ordered set of
+egress rules. Each rule, depending on the `action` field of the rule, allows
+or drops traffic which matches both the `to` and `ports` sections. The example
+policy contains a single rule, which drops matched traffic on a single port,
+to the 10.0.10.0/24 subnet specified by the `ipBlock` field.
+**Note**: The order in which the egress rules are set matter. i.e. rules will be
+evaluated in the order in which they are written.
+
+## Rule evaluation based on priorities
+
+Rules belonging to Cluster NetworkPolicy CRDs are associated with various
+priorities, such as the `priority` at the CNP level and the priority at rule
+level. Overall, Cluster Policy with highest precedence (lowest priority number
+value) is evaluated first. Within this policy, rules are evaluated in the order
+in which they are set. For example, consider the following:
+
+- CNP1{priority: 10, ingressRules: [ir1.1, ir1.2], egressRules: [er1.1, er1.2]}
+- CNP2{priority: 15, ingressRules: [ir2.1, ir2.2], egressRules: [er2.1, er2.2]}
+
+This translates to the following order:
+- Ingress rules: ir1.1 -> ir1.2 -> ir2.1 -> ir2.2
+- Egress rules: er1.1 -> er1.2 -> er2.1 -> er2.2
+
+Once a rule is matched, it is executed based on the action set.
+
+## Behavior of `to` and `from` selectors
+
+There are four kinds of selectors that can be specified in an ingress from
+section or egress to section:
```

Comment:
for this whole section, use `ingress` and `egress` for clarity I think

Reply:
done